### PR TITLE
Config files

### PR DESCRIPTION
**Pull request by randyhucker-physna**
Don't approve this please - non-functional

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -17,13 +17,14 @@
 import bpy.utils.previews
 from urllib.request import urlopen
 import os
+import requests
```
Comment (RandyHucker) [inline __init__.py:20]:
```suggestion
```

File: __init__.py [hunk 2]
```diff
@@ -17,13 +17,14 @@
...
 import os
+import requests
 from .thangs_fetcher import ThangsFetcher
 from .thangs_events import ThangsEvents
 from . import addon_updater_ops
 import socket
 import platform
-import requests
 import logging
+import configparser
```
Comment (RandyHucker) [inline __init__.py:27]:
```suggestion
```

File: thangs_events.py [hunk 3]
```diff
@@ -1,15 +1,28 @@
+from itertools import product
 import threading
+import configparser
 import requests
 import threading
 import time
 import logging
+import os
+
+# __location__ = os.path.realpath(
```
Comment (RandyHucker) [inline thangs_events.py:10]:
```suggestion
```

File: thangs_events.py [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
 import threading
+import configparser
 import requests
 import threading
 import time
 import logging
+import os
+
+# __location__ = os.path.realpath(
+#     os.path.join(os.getcwd(), os.path.dirname(__file__)))
```
Comment (RandyHucker) [inline thangs_events.py:11]:
```suggestion
```

File: thangs_events.py [hunk 5]
```diff
@@ -1,15 +1,28 @@
...
+# config_file = open(os.path.join(__location__, 'configfile.ini'))
```
Comment (RandyHucker) [inline thangs_events.py:12]:
```suggestion
```

File: dev_config.ini [hunk 6]
```diff
@@ -0,0 +1,3 @@
+[thangs]
+url = https://development.thangs.com/
+event_url = https://dev-events.thangs.com/
```
Comment (RandyHucker) [inline dev_config.ini:3]:
```suggestion
event_url = https://dev-events.thangs.com/

```

File: stage_config.ini [hunk 7]
```diff
@@ -0,0 +1,3 @@
+[thangs]
+url = https://staging.thangs.com/
+event_url = https://staging-events.thangs.com/
```
Comment (RandyHucker) [inline stage_config.ini:3]:
```suggestion
event_url = https://staging-events.thangs.com/

```

File: thangs_fetcher.py [hunk 8]
```diff
@@ -24,10 +25,15 @@
...
 from bpy.app.handlers import persistent
+import socket
 
 amplitude = ThangsEvents()
 amplitude.deviceId = socket.gethostname().split(".")[0]
 
+config_obj = configparser.ConfigParser()
+config_path = os.path.join(os.path.dirname(__file__), 'prod_config.ini')
+config_obj.read(config_path)
+thangs_config = config_obj['thangs']
```
Comment (craigv-physna) [inline thangs_fetcher.py:36]: Can we take this little snippet, move it into its own `config.py`, and then import `thangs_config` from `config.py` in both here and the events class?  It'll take out that little bit of duplication that we have in those classes

File: __init__.py [hunk 9]
```diff
@@ -578,7 +577,7 @@ def startSearch(self, value):
 def heartbeat_timer():
     log.info('sending thangs heartbeat')
     amplitude.send_amplitude_event("Thangs Blender Addon - Heartbeat", event_properties={'device_os': str(
-        fetcher.devideOS), 'device_ver': str(fetcher.deviceVer), 'source': "blender"})
+        fetcher.devideOS), 'device_ver': str(fetcher.deviceVer), 'source': "blender", 'addon_version': str(bl_info["version"])})
```
Comment (craigv-physna) [inline __init__.py:580]: Let's go ahead and clean things up a bit by sending `addon_version` in every amplitude event.  You should be able to assign it as a new field on the thangs_events class itself, and then let that class send it in with every amplitude event.

File: thangs_fetcher.py [hunk 10]
```diff
@@ -69,6 +68,14 @@ def __init__(self, callback=None):
         self.searching = False
         self.failed = False
         self.newSearch = False
+
+        
+        self.Thangs_Config = ThangsConfig()
```
Comment (craigv-physna) [inline thangs_fetcher.py:73]: Minor: I think we can change this to `get_config()`

**Review by craigv-physna — APPROVED** [review]

**Pull request merged**